net-test/bin/faucet.sh

```diff
@@ -0,0 +1,294 @@
+}
+
+http_401() {
+   printf "HTTP/1.0 401 Unsupported Method\r\n\r\n"
+}
+
+http_500() {
+   local ERR="$1"
+   local ERR_LEN=${#ERR}
+   printf "HTTP/1.0 500 Internal Server error\r\nContent-Length: $ERR_LEN\r\nContent-Type: text/plain\r\n\r\n$ERR"
+}
+
+http_404() {
+   printf "HTTP/1.0 404 Not Found\r\n\r\n"
+}
+
+get_ping() {
+   http_ok 5 "text/plain"
+   printf "alive"
+   return 0
+}
+
+get_bitcoin_ping() {
+   bitcoin-cli -conf="$BITCOIN_CONF" ping >/dev/null 2>&1
+   if [ $? -eq 0 ]; then 
+      local MSG="Bitcoind appears to be running"
+      http_ok ${#MSG} "text/plain"
+      echo "$MSG"
+      return 0
+   else
+      http_500 "Bitcoind appears to be stopped"
+      return 1
+   fi
+}
+
+get_balance() {
+   BALANCE="$(bitcoin-cli -conf="$BITCOIN_CONF" getbalance 2>&1)"
+   if [ $? -eq 0 ]; then
+      http_ok "${#BALANCE}" "text/plain"
+      echo "$BALANCE"
+      return 0
+   else
+      http_500 "$BALANCE"
+      return 1
+   fi
+}
+
+get_utxos() {
+   local ADDR="$1"
+   UTXOS="$(bitcoin-cli -conf="$BITCOIN_CONF" listunspent 1 1000000 "[\"$ADDR\"]" 2>&1)"
+   if [ $? -eq 0 ]; then 
+      http_ok ${#UTXOS} "application/json"
+      echo "$UTXOS"
+      return 0
+   else
+      http_500 "$UTXOS"
+      return 1
+   fi
+}
+
+get_confirmations() {
+   local TXID="$1"
+   CONFIRMATIONS="$(bitcoin-cli -conf="$BITCOIN_CONF" gettransaction "$TXID" | jq -r '.confirmations')"
+   RC=$?
+   if [ $RC -eq 0 ]; then 
+      http_ok ${#CONFIRMATIONS} "text/plain"
+      echo "$CONFIRMATIONS"
+      return 0
+   elif [ $RC -eq 1 ]; then 
+      http_500 "$CONFIRMATIONS"
+      return 1
+   else
+      http_404
+      return 2
+   fi
+}
+
+post_sendbtc() {
+   local ADDR
+
+   # format: address\n
+   read ADDR
+   TXID="$(bitcoin-cli -conf="$BITCOIN_CONF" sendtoaddress "$ADDR" "$FAUCET_AMOUNT" 2>&1)"
+   if [ $? -ne 0 ]; then
+      http_500 "$TXID"
+      return 1
+   fi
+
+   ERR="$(bitcoin-cli -conf="$BITCOIN_CONF" importaddress "$ADDR" 2>&1)"
+   if [ $? -ne 0 ]; then
+      http_500 "$ERR"
+      return 1
+   fi
+
+   http_ok ${#TXID} "text/plain"
+   echo "$TXID"
+   return 0
+}
+
+parse_request() {
+   local REQLINE
+   local VERB=""
+   local REQPATH=""
+   local CONTENT_TYPE=""
+   local CONTENT_LENGTH=0
+
+   while read REQLINE; do
+      # trim trailing whitespace
+      REQLINE="${REQLINE%"${REQLINE##*[![:space:]]}"}"
+      if [ -z "$REQLINE" ]; then
+         break
+      fi
+
+      # log "   reqline = '$REQLINE'"
+
+      TOK="$(echo "$REQLINE" | egrep "GET|POST" | sed -r 's/^(GET|POST)[ ]+([^ ]+)[ ]+HTTP\/1.(0|1)$/\1 \2/g')"
+      if [ -n "$TOK" ] && [ -z "$VERB" ] && [ -z "$REQPATH" ]; then 
+         set -- $TOK
+         VERB="$1"
+         REQPATH="$2"
+         continue
+      fi
+
+      TOK="$(echo "$REQLINE" | grep -i "content-type" | cut -d ' ' -f 2)"
+      if [ -n "$TOK" ] && [ -z "$CONTENT_TYPE" ]; then
+         CONTENT_TYPE="${TOK,,}"
```

Comment:
this syntax is not supported on mac os default bash (3.x), this is a 4.x syntax.  Instead I used
```
CONTENT_TYPE="$( tr 'A-Z' 'a-z' <<<"$TOK" )"
```

Reply:
The most recent versions of GNU bash are 5.x.x.x.  Not sure why Apple is punishing its users with extremely out-dated tools.

Reply:
I updated the code to exit if the version of bash is lower than 4.x.  I don't have an old version of bash handy do test with, and I don't think it's a good use of time to try and support (especially since the deployment method for these tools is ultimately going to be within a Docker container, which will be guaranteed to have a recent version of bash).

Reply:
my understanding is that Apple now favors `zsh` but too many things would break if they removed bash